extern/boostd-data/client/client.go

```diff
@@ -27,6 +27,7 @@ type Store struct {
 		GetOffsetSize             func(context.Context, cid.Cid, mh.Multihash) (*model.OffsetSize, error)
 		ListPieces                func(ctx context.Context) ([]cid.Cid, error)
 		PiecesCount               func(ctx context.Context, maddr address.Address) (int, error)
+		ScanProgress              func(ctx context.Context, maddr address.Address) (*types.ScanProgress, error)
```

Comment:
The API for boostd-data has exploded quite a bit with auxiliary functionality.

I wonder if we can refactor in the future and split into 2 APIs -- one for critical functionality that one needs to implement if they want their own adapter for a new database (such as PostgreSQL with Citus), and one for auxiliary functionality that we build - Piece Doctor, UI feedback (such as ScanProgress), etc.

Just a thought as we keep on adding new methods here, no action required at this time.

Reply:
That's true, we are adding a bunch of new methods 👍